code/game/machinery/_machinery.dm

```diff
@@ -955,11 +955,39 @@
 	return TRUE
 
 /obj/machinery/proc/display_parts(mob/user)
-	. = list()
-	. += span_notice("It contains the following parts:")
-	for(var/obj/item/C in component_parts)
-		. += span_notice("[icon2html(C, user)] \A [C].")
-	. = jointext(., "")
+	//the initial value for an particular part/stack type. We count up from here when more instances are found
+	var/init_value = 0
+	var/list/part_count = list()
+	for(var/obj/item/component_part in component_parts)
+		if(!part_count[component_part.name])
+			if(isstack(component_part))
+				var/obj/item/stack/stack_part = component_part
+				init_value = stack_part.amount
+			else
+				init_value = 1
+			part_count[component_part.name] = init_value
+		else
+			part_count[component_part.name]++
+
+	var/count = 0
+	var/suffix = ""
+	var/list/printed_components = list()
+
+	var/text = span_notice("It contains the following parts:")
+	for(var/obj/item/component_part in component_parts)
+		if(!printed_components[component_part.name])
+			count=part_count[component_part.name]
+			if(count>1)
+				if(!istype(component_part,/obj/item/stack/cable_coil) && isstack(component_part))
+					suffix = " sheets"
+				else
+					suffix = "s"
+			else
+				suffix=""
+			text+= span_notice("[icon2html(component_part, user)] [count] [component_part.name][suffix].")
```

Comment:
You didn't apply my change here

Reply:
oooffff

Reply:
You still didn't apply my change here :(

Reply:
Small question.  If I remove the check for cable coil then it would print 10 glasses,  10 plasteels and not 10 glass sheets, 10 plasteel sheets. 
That's looks a little silly is that okay?

Reply:
Anyway as u wish

Reply:
![Screenshot (49)](https://user-images.githubusercontent.com/110812394/203708005-1e6d8f22-179e-4a44-a681-de62d2f3ea87.png)

This looks wack but alright

Reply:
Or better yet i'll remove the suffix altogether that will fix it

Reply:
![Screenshot (50)](https://user-images.githubusercontent.com/110812394/203709212-ed4f8b76-0a9b-42b7-8f5e-397aed7b32b1.png)

Alright this looks ok too. I'll just go with this